war/src/main/js/add-item.js

```diff
@@ -0,0 +1,293 @@
+      var $tab = $(['<li><a class="tab" href="#',cleanHref(elem.id),'">',elem.name,'</a></li>'].join(''))
+        .click(function(){
+          //e.preventDefault(e);
+          var $this = $(this).children('a');
+          
+          var tab = $this.attr('href');
+          var scrollTop = $(cleanHref(tab,true)).offset().top - ($newView.children('.jenkins-config-widgets').height() + 15);
+          
+          setTimeout(function(){resetActiveTab($this);},510);
+          
+          $('html,body').animate({
+            scrollTop: scrollTop
+          }, 500);
+        });
+      return $tab;
+    }
+
+    function drawCategory(i,elem){
+      if (!elem) elem = i;
+      var $category = $('<div/>').addClass('category jenkins-config hide-cat').attr('id', 'j-add-item-type-'+elem.id);
+      var $items = $('<ul/>').addClass('j-item-options').appendTo($category);
+      var $newTarget;
+      
+      if(checkCatCount(elem)){
+        var $catHeader = $('<div class="category-header" />').prependTo($category);
+        var catDom = (elem.minToShow > 0)?
+            ['<h2>', elem.name, '</h2>'].join(''):
+              '';
+        $(catDom).appendTo($catHeader);
+        $(['<p>', elem.description, '</p>'].join('')).appendTo($catHeader);
+        
+        $category.removeClass('hide-cat');
+      }
+      else if(elem.remainders){
+        $newTarget = $('#'+cleanHref(elem.remainders,true)).find('.j-item-options');
+      }
+
+      return $items;
+    }
+    
+    function drawItem(elem){
+      var $item = $([
+          '<li class="',cleanClassName(elem.class),'"><label><input name="mode" value="',elem.class,'" type="radio" /> <span class="label">', elem.displayName, '</span></label></li>'
+      ].join('')).append([
+          '<div class="desc">', elem.description, '</div>'
+      ].join('')).append([
+          '<div class="icn"><span class="img" style="background:url(',jRoot,'/images/items/',cleanClassName(elem.class),'.png)"></span></div>'
+      ].join(''));
+
+      function setSelectState(){
+        var $this = $(this).closest('li');
+        //if this is a hyperlink, don't move the selection.
+        if($this.find('a:focus').length === 1) {return false;}
+        $this.closest('.categories').find('.active').removeClass('active');
+        $this.addClass('active');
+        $this.find('input[type="radio"]').attr('checked', 'checked');
+        window.updateOk($form[0]);
+        
+        $('html, body').animate({
+          scrollTop:$this.offset().top - 200
+        },50);
+        
+      }
+      
+      $item.click(setSelectState);
+      
+      return $item;
+    }
+    
+    // initialize
+    
+    var sortedDCategories = sortItemsByOrder(data.categories);
+    var sortedDCategoriesWithCopy = addCopyOption(sortedDCategories);
+    drawTabs(sortedDCategoriesWithCopy);
+    
+  });
+});
+
+function fireBottomStickerAdjustEvent() {
+  Event.fire(window, 'jenkins:bottom-sticker-adjust'); // jshint ignore:line
```

Comment:
Event.fire is part of Prototype.js. It is a means of attaching custom event listeners the the old Jenkins form used to stick the form submit controls at the bottom of the page, regardless of the window size. This is just a shim to trigger that legacy code.